src/Data/Entity/Position.php

```diff
@@ -5,10 +5,62 @@
 
 class Position extends Entity
 {
+    protected $accountID;
+    protected $name;
+    protected $color;
+    protected $sort;
 
-    protected $properties = [
-        'id', 'user_id', 'start_time', 'end_time',
-    ];
+    public function getAccountId()
+    {
+        return $this->accountID;
+    }
 
+    public function setAccountId($accountID)
+    {
+        $this->accountID = (int)$accountID;
+    }
 
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    public function setName($name)
+    {
+        $this->name = ucwords($name);
+    }
+
+    public function getColor()
+    {
+        return $this->color;
+    }
+
+    public function setColor($color)
+    {
+        $this->color = strtoupper($color);
+    }
+
+    public function getSort()
+    {
+        return $this->sort;
+    }
+
+    public function setSort($sort)
+    {
+        $this->sort = (int)$sort;
+    }
+
+    public function getData()
```

Comment:
I like the idea of using a generic method like this where can add a decorator to make this JSON serializable, instead add more cruft onto the entity.